pydantic/main.py

```diff
@@ -83,9 +83,9 @@ def __new__(
                     # in a single function
                     namespace['_init_private_attributes'] = _model_construction.init_private_attributes
 
-                    def __pydantic_post_init__(self_: Any, **kwargs: Any) -> None:
-                        self_._init_private_attributes()
-                        self_.model_post_init(**kwargs)
+                    def __pydantic_post_init__(self_: Any, context: Any) -> None:
```

Comment:
Here is another place where the signature is dictated by the fact that it needs to accept positional arguments and be callable via keyword from python. (Not sure if we'd prefer this signature or `*args` if it wasn't called with keyword arguments anywhere)